C - 3 - 4
C - 2 => should throw exception reading this line